src/javascript/crypto/e2e/openpgp/pgpmime/mimenode.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview main module for the safemail app.
+ * @author yzhu@yahoo-inc.com (Yan Zhu)
+ */
+
+goog.provide('e2e.openpgp.pgpmime.MimeNode');
+
+goog.require('e2e.openpgp.pgpmime.Constants');
+goog.require('e2e.openpgp.pgpmime.Text');
+goog.require('e2e.openpgp.pgpmime.Utils');
+
+goog.require('goog.array');
+goog.require('goog.asserts');
+goog.require('goog.crypt.base64');
+goog.require('goog.object');
+goog.require('goog.string');
+
+goog.scope(function() {
+var pgpmime = e2e.openpgp.pgpmime;
+var constants = pgpmime.Constants;
+
+
+
+/**
+ * Constructor for a MIME tree node.
+ * @param {{
+ *   multipart: boolean,
+ *   optionalHeaders: (Array<!e2e.openpgp.pgpmime.types.Header>|undefined)
+ * }} options Options to initialize for the node.
+ * @param {e2e.openpgp.pgpmime.MimeNode=} opt_parent The parent node.
+ * @constructor
+ */
+e2e.openpgp.pgpmime.MimeNode = function(options, opt_parent) {
+  this.parent = opt_parent || this;
+  this.multipart_ = options.multipart;
+
+  this.children_ = [];
+  this.header_ = /** @type {e2e.openpgp.pgpmime.types.Header} */ ({});
+  this.content_ = null;
```

Comment:
Please add @types to all those internal variables. /** @private {type} */ will do, although usually we do:
```
/** 
 * Description
 * @private {type}
 */
this.var_ = xxx;
```